network/validator/senderValidator.go

```diff
@@ -15,14 +15,20 @@ type SenderValidator struct {
 	sender []byte
 }
 
-// NewSenderValidator creates and returns a new SenderValidator for the given sender ID
-func NewSenderValidator(sender flow.Identifier) *SenderValidator {
+// ValidateSender creates and returns a new SenderValidator for the given sender ID
+func ValidateSender(sender flow.Identifier) network.MessageValidator {
 	sv := &SenderValidator{}
 	sv.sender = sender[:]
 	return sv
 }
 
-// Validate returns true if the message origin id is different from the sender ID.
+// Validate returns true if the message origin id is the same as the sender ID.
```

Comment:
The name `SenderValidator` implies that the validator checks if the sender ID  is as expected. Hence changing the `!Equals` to `Equals`.